solr/core/src/java/org/apache/solr/cloud/LeaderElector.java

```diff
@@ -131,14 +136,34 @@ private void checkIfIamLeader(final ElectionContext context, boolean replacement
     }
 
     if (leaderSeqNodeName.equals(seqs.get(0))) {
-      // I am the leader
+      // I am the leader - watch my node in case leadership is somehow revoked
+      try {
+        ElectionWatcher watcher = new ElectionWatcher(path, path, getSeq(path), context);
+        if (zkClient.exists(path, watcher, true) == null) {
```

Comment:
There's a super weird case to think through here.  `exists` can be problematic because it really does two different things:
1) If the node exists, leave a watch that triggers on deletion
2) If the node does not exist, leave a watch that triggers on creation

The first feature you want; the second feature you almost certainly don't.  It might be more solid here to to do a getData() call instead and catch NoNodeExeception, so that you only leave a watch on the success case, and don't leave a dangling watch on the missing node case.